internal/prometheusbp/spectest/spec.go

```diff
@@ -142,22 +145,42 @@ func validateLabels(name, prefix, clientOrServer string, gotLabels map[string]st
 
 // buildLables returns a set of expected labels for the metric name provided.
 // prefix is either thrift, http, or grpc.
+func buildLables(name, prefix, clientOrServer string) map[string]struct{} {
+	var labelSuffixes = []string{}
+	switch prefix {
+	case thriftPrefix:
+		labelSuffixes = thriftSpecificLabels(name)
+	case grpcPrefix:
+		labelSuffixes = grpcSpecificLabels(name)
+	default:
+		labelSuffixes = []string{}
```

Comment:
this is a no-op, and it should be `= nil` since "allocating" a zero slice is unnecessary.

Probably just drop the default